static/app/views/performance/transactionSummary/transactionEvents/content.tsx

```diff
@@ -137,65 +125,74 @@ function Search(props: Props) {
   );
 
   return (
-    <SearchWrapper>
+    <FilterActions>
       <Filter
         organization={organization}
         currentFilter={spanOperationBreakdownFilter}
         onChangeFilter={onChangeSpanOperationBreakdownFilter}
       />
+      {organization.features.includes('selection-filters-v2') && (
+        <PageFilterBar condensed>
+          <EnvironmentPageFilter />
+          <DatePageFilter alignDropdown="left" />
+        </PageFilterBar>
+      )}
       <StyledSearchBar
         organization={organization}
         projectIds={eventView.project}
         query={query}
         fields={eventView.fields}
         onSearch={handleSearch}
       />
-      <SearchRowMenuItem>
-        <DropdownControl
-          buttonProps={{prefix: t('Percentile')}}
-          label={eventsFilterOptions[eventsDisplayFilterName].label}
-        >
-          {Object.entries(eventsFilterOptions).map(([name, filter]) => {
-            return (
-              <DropdownItem
-                key={name}
-                onSelect={onChangeEventsDisplayFilter}
-                eventKey={name}
-                data-test-id={name}
-                isActive={eventsDisplayFilterName === name}
-              >
-                {filter.label}
-              </DropdownItem>
-            );
-          })}
-        </DropdownControl>
-      </SearchRowMenuItem>
-      <SearchRowMenuItem>
-        <Button to={eventView.getResultsViewUrlTarget(organization.slug)}>
-          {t('Open in Discover')}
-        </Button>
-      </SearchRowMenuItem>
-    </SearchWrapper>
+      <DropdownControl
+        buttonProps={{prefix: t('Percentile')}}
+        label={eventsFilterOptions[eventsDisplayFilterName].label}
+      >
+        {Object.entries(eventsFilterOptions).map(([name, filter]) => {
+          return (
+            <DropdownItem
+              key={name}
+              onSelect={onChangeEventsDisplayFilter}
+              eventKey={name}
+              data-test-id={name}
+              isActive={eventsDisplayFilterName === name}
+            >
+              {filter.label}
+            </DropdownItem>
+          );
+        })}
+      </DropdownControl>
+      <Button to={eventView.getResultsViewUrlTarget(organization.slug)}>
+        {t('Open in Discover')}
+      </Button>
+    </FilterActions>
   );
 }
 
-const SearchWrapper = styled('div')`
-  display: flex;
-  width: 100%;
-  margin-bottom: ${space(3)};
-`;
+const FilterActions = styled('div')`
+  display: grid;
+  gap: ${space(2)};
+  margin-bottom: ${space(2)};
 
-const StyledSearchBar = styled(SearchBar)`
-  flex-grow: 1;
-`;
+  @media (min-width: ${p => p.theme.breakpoints[0]}) {
+    grid-template-columns: repeat(4, min-content);
```

Comment:
we may need to have a different number of columns based on whether or not the page filters are shown due to the feature flagging. This is what the page looks like if the feature flag is turned off which would be true for self hosted and ST 
<img width="1186" alt="image" src="https://user-images.githubusercontent.com/9372512/165816684-ddceb8a5-3659-493f-8388-1479aa312e3c.png">

Reply:
This probably applies to the other pages too where grid is now used

Reply:
Wait I thought this was in GA?

Reply:
@doralchan I think the feature flag is controlled via flagr. And it could be disabled for some orgs in case the GA needs to be partially reverted temporarily.

I think we could dynamically control the grid columns like this:

```suggestion
    grid-template-columns: ${p => p.organization.features.includes('selection-filters-v2') ? 'repeat(4, min-content)' : 'repeat(3, min-content)'};
```

Reply:
Yup something like that would suffice, though it is quite verbose so I don't mind trying to remove the feature flag logic today and then we can rebase off that change. Or maybe even easier would for me to fork off this branch and remove the feature flag logic, get that approved and merge back into this PR. Lmk if either of those would be preferred to adding the feature flag logic into the styled components.

Reply:
@davidenwang removing the conditional feature flag logic would be better 👍

Reply:
doing it here https://github.com/getsentry/sentry/pull/34081

Reply:
@davidenwang Amazing. 🌮